examples/server/utils.hpp

```diff
@@ -489,6 +489,25 @@ static size_t find_partial_stop_string(const std::string &stop, const std::strin
     return std::string::npos;
 }
 
+/* returns a pair containing the position and index of first element found.
+   returns npos and -1 if not found */
+static std::pair<size_t, int> find_first_substring(const std::string &haystack, const std::vector<std::string> & needles, size_t search_pos = 0)
```

Comment:
Hmm I didn't think much about the fact that we're supporting multiple start strings here.

In this case, probably returning `std::pair<size_t, std::string &>` would be a better choice.